### PR TITLE
Add Kanban board to ExampleApp & Improve the interfaces of JSONObject and JSONArray

### PR DESCRIPTION
**Pull request by hyeongsik-won**
**What this PR does / why we need it**:

Added Kanban board to ExampleApp and bound Document and KanbanViewModel.
Improved the interfaces of JSONObject and JSONArray.
Added additional test codes.
Added some utils.
Fixed some bugs in JSONArray and JSONObject.

The Kanban demo video

https://user-images.githubusercontent.com/12286308/200215704-b77b4cf2-c07b-4f49-ac46-192e9018f8d9.mp4

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

**Additional documentation**:

```docs

```

**Checklist**:
- [ ] Added relevant tests or not required
- [ ] Didn't break anything

### CODE REVIEW

File: Tests/Document/JSONObjectTests.swift [hunk 1]
```diff
@@ -155,4 +155,31 @@ class JSONObjectTests: XCTestCase {
...
+    }
+
+    struct JsonArrayTestType: YorkieJSONObjectable {
+        let id: Int64 = 200
+    }
+
+    func test_can_insert_obejct() async {
+        let target = Document(key: "doc1")
+        await target.update { root in
+            root.object = JsonObejctTestType()
```
Comment (hyeongsik-won) [inline Tests/Document/JSONObjectTests.swift:177]: Types confirming ``YorkieJSONObjectable``  can be easily injected into the document.

File: Tests/Document/JSONObjectTests.swift [hunk 2]
```diff
@@ -155,4 +155,31 @@ class JSONObjectTests: XCTestCase {
             XCTAssertEqual(idOfCompD, "d-2")
         }
     }
+
+    struct JsonObejctTestType: YorkieJSONObjectable {
+        var id: Int64 = 100
+        var type: String = "struct"
+        var serial: Int32 = 1234
+        var array: [JsonArrayTestType] = [JsonArrayTestType()]
```
Comment (hyeongsik-won) [inline Tests/Document/JSONObjectTests.swift:163]: Arrays including types confirming YorkieJSONObjectable can be easily injected into the document.

File: Sources/Document/Util/YorkieJSONObjectable.swift [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+public protocol YorkieJSONObjectable {
+    /// The members of ``excludedLabels`` is not included in a dictionary made by ``toYorkieObject``.
+    var excludedLabels: [String] { get }
+}
+
+public extension YorkieJSONObjectable {
+    /// ``toJsonObject`` make a dictionary including members of a type confirming ``YorkieObjectable``.
+    internal var toJsonObject: [String: Any] {
+        var result = [String: Any]()
+        Mirror(reflecting: self).children.forEach { child in
```
Comment (humdrum) [inline Sources/Document/Util/YorkieJSONObjectable.swift:29]: How about applying Codable?

**Review by humdrum — APPROVED** [review]
LGTM!

File: Sources/Document/Util/YorkieJSONObjectable.swift (hunk 3 above)
Comment (hyeongsik-won) [reply Sources/Document/Util/YorkieJSONObjectable.swift]: I have changed ``YorkieJSONObjectable`` from ``Mirrot`` to Codable.

**Pull request merged**